### PR TITLE
client: use appropriate ExecutionContext for initial sync / regular import

### PR DESCRIPTION
**Pull request by andresilva**
We were not making the distinction when importing blocks and just using the default "other" execution context.

### CODE REVIEW

File: client/service/src/client/client.rs [hunk 1]
```diff
@@ -865,9 +861,15 @@ impl<B, E, Block, RA> Client<B, E, Block, RA> where
 			// block.
 			(true, ref mut storage_changes @ None, Some(ref body)) => {
 				let runtime_api = self.runtime_api();
+				let execution_context = if import_block.origin == BlockOrigin::NetworkInitialSync {
```
Comment (bkchr) [inline client/service/src/client/client.rs:864]: I think we should follow the comment in the source for the context:
```
	/// Context for general importing (including own blocks).
	Importing,
	/// Context used when syncing the blockchain.
	Syncing,
```

Comment (andresilva) [reply client/service/src/client/client.rs:864]: I assumed that `Importing` is for any block import (not only for own blocks). This is the same distinction we make in the informant when we print `Syncing` or `Imported`, the only difference is the queue size.

Comment (bkchr) [reply client/service/src/client/client.rs:864]: But we always sync :D Even after the initial sync. The point is that this is a security consideration (not trusting foreign blocks).

**Comment by andresilva** [comment]
Updated the `ExecutionContext` variants to distinguish between "own block import" and "foreign block import" (rather than import / sync).

File: client/cli/src/params/import_params.rs [hunk 2]
```diff
@@ -132,25 +134,25 @@ impl ImportParams {
 /// Execution strategies parameters.
 #[derive(Debug, StructOpt, Clone)]
 pub struct ExecutionStrategiesParams {
-	/// The means of execution used when calling into the runtime while syncing blocks.
+	/// The means of execution used when calling into the runtime while import locally authored blocks.
 	#[structopt(
-		long = "execution-syncing",
+		long = "execution-own-block-import",
```
Comment (cecton) [inline client/cli/src/params/import_params.rs:139]:
```suggestion
		long,
```
The name is automatically generated based on the field already

File: client/cli/src/params/import_params.rs [hunk 3]
```diff
@@ -132,25 +134,25 @@ impl ImportParams {
...
+		default_value = DEFAULT_EXECUTION_OWN_BLOCK_IMPORT.as_str(),
 	)]
-	pub execution_syncing: ExecutionStrategy,
+	pub execution_own_block_import: ExecutionStrategy,
 
-	/// The means of execution used when calling into the runtime while importing blocks.
+	/// The means of execution used when calling into the runtime while importing foreign blocks.
 	#[structopt(
-		long = "execution-import-block",
+		long = "execution-foreign-block-import",
```
Comment (cecton) [inline client/cli/src/params/import_params.rs:149]:
```suggestion
		long,
```

**Comment by andresilva** [comment]
Reverted to keep the original semantics and extended docs with your comment. I still kept the default execution context for regular import as `NativeElseWasm`, do you think we should change it to `Wasm` or keep that at the discretion of validators who want to be extra safe?

**Comment by gavofyork** [comment]
> Reverted to keep the original semantics and extended docs with your comment. I still kept the default execution context for regular import as `NativeElseWasm`, do you think we should change it to `Wasm` or keep that at the discretion of validators who want to be extra safe?

Non-validating nodes should be entirely `NativeElseWasm`. Validator nodes should be `Wasm` when doing keep-in-sync importing including when importing their own blocks (but *not* when authoring them). In other circumstances, it should be `NativeElseWasm`.

**Pull request merged**